### ISSUE TITLE
[Bug]: dApps or Websites: Console Errors in lockdown-run.js and lockdown-more.js for MetaMask v.11.0.0

### ISSUE THREAD

**Issue by Muurtaja** [MetaMask/metamask-extension#21135]
### Describe the bug

Encountered a console error while using both web3-based and regular websites. The error disappears upon uninstalling Metamask. This issue is causing confusion and needs attention.

### Expected behavior

Metamask installed in Chrome should not trigger any console errors on web3-based or other websites.

### Screenshots

![image](https://github.com/MetaMask/metamask-extension/assets/42427285/9d23d1e2-101f-40a3-af73-aa95e0a0b2cf)

### Steps to reproduce

1. Install Metamask
2. Visit a web3-based or regular website.
3. Observe the console for any errors.
4. Uninstall Metamask.
5. Repeat step 1.
6. Verify if the console error persists.

### Error messages or log output

```shell
lockdown-run.js:17 Lockdown failed: TypeError: At intrinsics.Object.groupBy expected boolean not function
  at isAllowedPropertyValue (lockdown-install.js:1:53384)
  at isAllowedProperty (lockdown-install.js:1:53807)
  at visitProperties (lockdown-install.js:1:55095)
  at isAllowedPropertyValue (lockdown-install.js:1:53041)
  at isAllowedProperty (lockdown-install.js:1:53807)
  at visitProperties (lockdown-install.js:1:55095)
  at lockdown-install.js:1:55523
  at repairIntrinsics (lockdown-install.js:1:144597)
  at lockdown-install.js:1:145462
  at lockdown-run.js:4:3
(anonymous) @ lockdown-run.js:17
Show 1 more frame
Show less
lockdown-more.js:99 Protecting intrinsics failed: ReferenceError: harden is not defined
  at lockdown-more.js:69:13
  at Set.forEach (<anonymous>)
  at protectIntrinsics (lockdown-more.js:44:22)
  at lockdown-more.js:97:5
(anonymous) @ lockdown-more.js:99
Show 1 more frame
Show less
lockdown-run.js:17 Lockdown failed: TypeError: At intrinsics.Object.groupBy expected boolean not function
  at isAllowedPropertyValue (lockdown-install.js:1:53384)
  at isAllowedProperty (lockdown-install.js:1:53807)
  at visitProperties (lockdown-install.js:1:55095)
  at isAllowedPropertyValue (lockdown-install.js:1:53041)
  at isAllowedProperty (lockdown-install.js:1:53807)
  at visitProperties (lockdown-install.js:1:55095)
  at lockdown-install.js:1:55523
  at repairIntrinsics (lockdown-install.js:1:144597)
  at lockdown-install.js:1:145462
  at lockdown-run.js:4:3
(anonymous) @ lockdown-run.js:17
Show 1 more frame
Show less
lockdown-more.js:99 Protecting intrinsics failed: ReferenceError: harden is not defined
  at lockdown-more.js:69:13
  at Set.forEach (<anonymous>)
  at protectIntrinsics (lockdown-more.js:44:22)
  at lockdown-more.js:97:5
```

### Version

11.0.0

### Build type

None

### Browser

Chrome

### Operating system

Windows

### Severity

Clients are experiencing significant confusion and apprehension after downloading Metamask, suspecting potential asset theft. This has resulted in discouragement towards using our dApp, which could lead to a loss of user trust and adoption. Immediate attention is required to address this critical issue.

**Comment by crypblizz8** [comment]
Experiencing same issue with the above reproduction steps. 

Please look into this as MM Extension, should not be loading errors on immediate load.

**Comment by Muurtaja** [comment]
Need a concern  regarding this. 🥲

**Comment by anaamolnar** [comment]
I am closing this in favor of https://github.com/MetaMask/metamask-extension/issues/20847

**Comment by Muurtaja** [comment]
The issue has been resolved in version 11.1.2. Thank you all for your assistance!
